src/fermilib/utils/_unitary_cc.py

```diff
@@ -18,6 +18,7 @@
 
 from fermilib.ops import FermionOperator
 from fermilib.transforms import jordan_wigner
+from functools import partial
```

Comment:
True fact, I replaced that with lambda functions instead.  I think MainEngine might also be unused in the test file, which I can also fix.

I'll also be the first to note that some of these ProjectQ filter and mapper functions should probably find a home outside of _unitary_cc.py.  They are a bit more general even though that's where they are finding the most use at the moment.